src/data/list/zip.lean

```diff
@@ -287,4 +301,62 @@ begin
     { simp [hl, mul_add] } }
 end
 
+section distrib
```

Comment:
A brief `/!- ### heading -/` comment summarizing the obvious pattern here "operations can be applied before or after a zip" might be good for doc-gen